13 ShouldUpdate/src/face.tsx

```diff
@@ -30,19 +30,11 @@ export class FaceComponent extends React.Component<Props, {}> {
   {
     const rangeChange = [100, 200, 300, 400];
 
-    let index =  0;
-    let isRangeChange = false;
-
-    while(!isRangeChange && index < rangeChange.length) {
-      isRangeChange = (this.props.level < rangeChange[index] && nextProps.level >= rangeChange[index])
-                    ||
-                      (this.props.level > rangeChange[index] && nextProps.level <= rangeChange[index])
-      ;
-
-      index++;
-    }
-
-     return isRangeChange;
+    // some returns true if any pass the check.
+    return rangeChange.some(elem => {
```

Comment:
Great work, It's a nice change. To go one step over, what do you think to extract this method in other file.

Some thing like:

```javascript
const ranges = [100, 200, 300, 400];

export function isThereSomeRangeTransition (currentLevel, nextLevel) {
  return ranges.some((range) => {
    return isAnUpperRangeTransition(currentLevel, nextLevel, range) ||
      isALowerRangeTransition(currentLevel, nextLevel, range);
  });
}

function isAnUpperRangeTransition(currentLevel, nextLevel, range) {
  return currentLevel < range &&
    nextLevel >= range;
}

function isALowerRangeTransition(currentLevel, nextLevel, range) {
  return currentLevel > range &&
    nextLevel <= range;
}
```